### PR TITLE
feat(pattern): added CTA updated fun specs and dev ontology refactor

### PR DESCRIPTION
**Pull request by mkothur**
### Related Ticket(s)

Pattern: ContentGroup - #1173 

### Description

Added the basic grid in story to look nicer and easy testing. &
Added CTA component with card style and local type.
Updated the README file and removed the featureFlags.
image

### Changelog

**New**

**Changed**

Patterns- sub-pattern: ContentGroup.

### CODE REVIEW

File: packages/react/src/patterns/sub-patterns/ContentGroup/__stories__/ContentArrayGroup.stories.js [hunk 1]
```diff
@@ -14,10 +12,23 @@ storiesOf('Patterns (Sub-Patterns)|ContentGroup', module)
...
+      title: renderCTA && text('CTA heading', 'Lorem ipsum dolor sit amet'),
+      href: renderCTA && text('href', 'https://ibm.com'),
+    };
 
-    return <ContentGroup heading={copy} />;
+    return (
+      <div className="bx--grid ">
+        <div className="bx--row">
+          <div className="bx--col-sm-4 bx--col-lg-8 bx--offset-lg-4">
+            <ContentGroup heading={heading} {...(renderCTA && { ...cta })} />
```
Comment (annawen1) [inline packages/react/src/patterns/sub-patterns/ContentGroup/__stories__/ContentArrayGroup.stories.js]: `{...(renderCTA && { ...cta })}` looks weird

can we do `{renderCTA &&  ...cta}` instead?

Comment (mkothur) [reply packages/react/src/patterns/sub-patterns/ContentGroup/__stories__/ContentArrayGroup.stories.js]: Okay.

File: packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js [hunk 2]
```diff
@@ -20,41 +21,44 @@ const { prefix } = settings;
...
-              {heading}
-            </h3>
-          </div>
+const ContentGroup = ({ children, heading, ...cta }) => (
+  <div
+    className={`${prefix}--content-group`}
+    data-autoid={`${stablePrefix}--content-group`}>
+    <div className={`${prefix}--content-group__container`}>
+      <div className={`${prefix}--content-group__row`}>
+        <div className={`${prefix}--content-group__col`}>
```
Comment (photodow) [inline packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: I don't think these are doing anything? Am I missing it?

File: packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js [hunk 3]
```diff
@@ -20,41 +21,44 @@ const { prefix } = settings;
...
+          <h3
+            data-autoid={`${stablePrefix}--content-group__title`}
+            className={`${prefix}--content-group__title`}>
+            {heading}
+          </h3>
+        </div>
+      </div>
+      <div className={`${prefix}--content-group__row`}>
```
Comment (photodow) [inline packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: I don't see this doing anything.

File: packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js [hunk 4]
```diff
@@ -20,41 +21,44 @@ const { prefix } = settings;
...
+        <div
+          data-autoid={`${stablePrefix}--content-group__children`}
+          className={classNames(
+            `${prefix}--content-group__col`,
+            `${prefix}--content-group__children`
+          )}>
+          {children}
         </div>
         <div className={`${prefix}--content-group__row`}>
```
Comment (photodow) [inline packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: I don't see this doing anything.

File: packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js [hunk 5]
```diff
@@ -20,41 +21,44 @@ const { prefix } = settings;
...
-      <div className={`${prefix}--content-group__container`}>
-        <div className={`${prefix}--content-group__row`}>
-          <div className={`${prefix}--content-group__col`}>
-            <h3
-              data-autoid={`${stablePrefix}--content-group__title`}
-              className={`${prefix}--content-group__title`}>
-              {heading}
-            </h3>
-          </div>
+const ContentGroup = ({ children, heading, ...cta }) => (
```
[the remark this answers is not in the document]
Comment (mkothur) [reply packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: In the functional spec has this
image

File: packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js (hunk 4 above)
Comment (mkothur) [reply packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: if row is not their CTA component is rendering inside the contentGroup placeholder
image

Comment (mkothur) [reply packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: sorry if we remove `row` at 41 line .looks like above.

Comment (mkothur) [reply packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: but row is not doing anything at line number 50. will remove that.

File: packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (photodow) [reply packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: Right so the prop would be `cta`. Within that prop you pass a `CTA` object. Something like the following for the text local `CTA`.

```json
{
  "style": "text",
  "type": "local",
  "copy": "IBM Homepage",
  "href": "www.ibm.com"
}
```

When you use the spread syntax like you're doing here you're catching whatever you didn't specify within the props there.

File: packages/react/src/patterns/sub-patterns/ContentGroup/__stories__/index.scss [hunk 6]
```diff
@@ -1,6 +1 @@
 @import '../../../../../../styles/scss/patterns/sub-patterns/content-group/content-group';
-
-.#{$prefix}--content-group__children {
-  height: 300px;
-  background-color: #f4f4f4;
-}
```
[the remark this answers is not in the document]
Comment (mkothur) [reply packages/react/src/patterns/sub-patterns/ContentGroup/__stories__/index.scss]: removed here and added in `_content-group.scss`

File: packages/styles/scss/patterns/sub-patterns/content-group/_content-group.scss [hunk 7]
```diff
@@ -41,12 +37,12 @@
     }
 
     &__children {
-      padding-left: 0;
-      padding-right: 0;
+      height: 300px;
+      background-color: #f4f4f4;
+      margin-bottom: $carbon--layout-03;
```
[the remark this answers is not in the document]
Comment (mkothur) [reply packages/styles/scss/patterns/sub-patterns/content-group/_content-group.scss]: background-color is already their in index.scss and I removed their and added here, I extra added is margin-bottom

File: packages/react/src/patterns/sub-patterns/ContentGroup/README.md [hunk 8]
```diff
@@ -35,9 +43,16 @@ ReactDOM.render(<App />, document.querySelector('#app'));
 
 ## Props
 
-| Name      | Required | Data Type | Default Value | Description  |
-| --------- | -------- | --------- | ------------- | ------------ |
-| `heading` | NO       | String    | null          | Heading text |
+| Name      | Required | Data Type | Default Value | Description                                |
+| --------- | -------- | --------- | ------------- | ------------------------------------------ |
+| `heading` | NO       | String    | null          | Heading text                               |
+| `cta`     | NO       | CTA       | Card          | CTA type for more details see `CTA` below. |
```
[the remark this answers is not in the document]
Comment (mkothur) [reply packages/react/src/patterns/sub-patterns/ContentGroup/README.md]: yeah will add.

File: packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js (hunk 4 above)
Comment (photodow) [reply packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: When I remove the grid I get the following. I think there might be another problem there.



Take a look at the rest of my feedback, it might resolve itself.

File: packages/react/src/patterns/sub-patterns/ContentGroup/README.md [hunk 9]
```diff
@@ -22,9 +22,17 @@ import { ContentGroup } from '@carbon/ibmdotcom-react';
...
-
-  return <ContentGroup heading={copy}>//Content array goes here</ContentGroup>;
+  const heading = 'Lorem ipsum dolor sit amet.';
+  const cta = {
+    title: 'Lorem ipsum dolor sit amet',
+    href: 'https://ibm.com',
+  };
+
+  return (
+    <ContentGroup heading={heading} {...cta}>
```
[the remark this answers is not in the document]
Comment (mkothur) [reply packages/react/src/patterns/sub-patterns/ContentGroup/README.md]: Okay

File: packages/react/src/patterns/sub-patterns/ContentGroup/__stories__/index.scss (hunk 6 above)
[the remark this answers is not in the document]
Comment (photodow) [reply packages/react/src/patterns/sub-patterns/ContentGroup/__stories__/index.scss]: We don't want this to be apart of the sub-pattern itself. It's merely to help visual the children while in storybook.

File: packages/styles/scss/patterns/sub-patterns/content-group/_content-group.scss (hunk 7 above)
[the remark this answers is not in the document]
Comment (photodow) [reply packages/styles/scss/patterns/sub-patterns/content-group/_content-group.scss]: See https://github.com/carbon-design-system/ibm-dotcom-library/pull/1326#discussion_r376621547

File: packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (mkothur) [reply packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: Ok that make sense now.

File: packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js (hunk 4 above)
Comment (mkothur) [reply packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: where did you remove the grid?

Comment (photodow) [reply packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: Here is what my `JSX` looks like after I removed the grid.

```jsx
<div
  className={`${prefix}--content-group`}
  data-autoid={`${stablePrefix}--content-group`}>
    <h3
      data-autoid={`${stablePrefix}--content-group__title`}
      className={`${prefix}--content-group__title`}>
        {heading}
    </h3>
    <div
      data-autoid={`${stablePrefix}--content-group__children`}
      className={classNames(
        `${prefix}--content-group__col`,
        `${prefix}--content-group__children`
      )}>
      {children}
    </div>
    <CTA style="card" type="local" {...cta} />
</div>
```

Looking in the `CSS` I'm not seeing those class names `__container`, `__row`, or `__col` set to anything.

**Comment by mkothur** [comment]
@photodow: updated the PR for above comments.

File: packages/react/src/patterns/sub-patterns/ContentGroup/__stories__/ContentArrayGroup.stories.js [hunk 10]
```diff
@@ -1,6 +1,5 @@
 import './index.scss';
-
-import { text, withKnobs } from '@storybook/addon-knobs';
+import { boolean, text, withKnobs } from '@storybook/addon-knobs';
```
Comment (jeffchew) [inline packages/react/src/patterns/sub-patterns/ContentGroup/__stories__/ContentArrayGroup.stories.js]: Filename should be renamed to `ContentGroup.stories.js`

Comment (mkothur) [reply packages/react/src/patterns/sub-patterns/ContentGroup/__stories__/ContentArrayGroup.stories.js]: Okay.

File: packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js [hunk 11]
```diff
@@ -20,41 +21,44 @@ const { prefix } = settings;
...
+  <div
+    className={`${prefix}--content-group`}
```
[the remark this answers is not in the document]
Comment (kennylam) [reply packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: @mkothur something like this:
```suggestion
  const ContentGroup = ({ children, heading, customClassName, ...cta }) => {
    const className= cx(`${prefix}--content-group`, customClassName);
    
    return (
      <div
        className={className}
```

[the remark this answers is not in the document]
Comment (mkothur) [reply packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: Okay.

[the remark this answers is not in the document]
Comment (mkothur) [reply packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: @kennylam : updated the PR for above comment.

File: packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js [hunk 12]
```diff
@@ -20,41 +21,47 @@ const { prefix } = settings;
...
-const ContentGroup = ({ children, heading }) => {
+const ContentGroup = ({
+  children,
+  heading,
+  className: customClassName,
+  cta,
+}) => {
+  const className = classNames(`${prefix}--content-group`, {
+    [customClassName]: !!customClassName,
+  });
```
Comment (photodow) [inline packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: Any reason we couldn't just pass in `customClassName` instead of a whole object? What's the value of the object?

File: packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js [hunk 13]
```diff
@@ -20,41 +21,47 @@ const { prefix } = settings;
  * @param {object} props props object
  * @param {*} props.children JSX Components
  * @param {string} props.heading  Heading objects
+ * @param {string} props.className optional class to be applied to the containing node
+ * @param {*} props.cta CTA component props object
  * @returns {*} JSX ContentArrayGroup component
```
Comment (photodow) [inline packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: `ContentGroup` not `ContentArrayGroup`

File: packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js [hunk 14]
```diff
@@ -20,41 +21,47 @@ const { prefix } = settings;
  * @param {object} props props object
  * @param {*} props.children JSX Components
  * @param {string} props.heading  Heading objects
```
Comment (photodow) [inline packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: Heading objects I think is simple a a string, not an object.

File: packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js [hunk 15]
```diff
@@ -20,41 +21,47 @@ const { prefix } = settings;
...
     </div>
   );
 };
 
 ContentGroup.propTypes = {
   heading: PropTypes.string,
-  children: PropTypes.element,
+  children: PropTypes.object,
+  className: PropTypes.string,
+  cta: PropTypes.object,
```
Comment (photodow) [inline packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: Can we use `instanceOf(CTA)` here?

File: packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js [hunk 16]
```diff
@@ -20,41 +21,47 @@ const { prefix } = settings;
...
+      <div
+        data-autoid={`${stablePrefix}--content-group__children`}
+        className={classNames(
+          `${prefix}--content-group__col`,
+          `${prefix}--content-group__children`
+        )}>
+        {children}
+      </div>
+      <div className={`${prefix}--content-group__CTA`}>
+        <CTA style="card" type="local" {...cta} />
```
Comment (photodow) [inline packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: Is this optional? Do we ever render null if the `CTA` is no available?

File: packages/styles/scss/patterns/sub-patterns/content-group/_content-group.scss [hunk 17]
```diff
@@ -22,31 +20,40 @@
...
     &__children {
-      padding-left: 0;
-      padding-right: 0;
+      margin-bottom: $carbon--layout-03;
+    }
+
+    &__CTA {
+      @include carbon--breakpoint('sm') {
+        @include carbon--make-col(1, 2);
+      }
```
Comment (photodow) [inline packages/styles/scss/patterns/sub-patterns/content-group/_content-group.scss]: `sm` doesn't do anything.

File: packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js (hunk 11 above)
[the remark this answers is not in the document]
Comment (kennylam) [reply packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: @mkothur turns out you do not need the condition to check for `customClassName`. the `classNames` package will ignore it if it doesn't exist. so we can remove `[customClassName]: !!customClassName,`. sorry about that!
thanks @photodow

File: packages/styles/scss/patterns/sub-patterns/content-group/_content-group.scss [hunk 18]
```diff
@@ -22,31 +20,40 @@
...
 
     &__children {
-      padding-left: 0;
-      padding-right: 0;
+      margin-bottom: $carbon--layout-03;
+    }
+
+    &__CTA {
+      @include carbon--breakpoint('sm') {
+        @include carbon--make-col(1, 2);
```
Comment (photodow) [inline packages/styles/scss/patterns/sub-patterns/content-group/_content-group.scss]: Smallest breakpoint should be 4,4. Since it's a block I don't think you need this.

File: packages/styles/scss/patterns/sub-patterns/content-group/_content-group.scss [hunk 19]
```diff
@@ -22,31 +20,40 @@
...
+      }
+      @include carbon--breakpoint('md') {
+        @include carbon--make-col(2, 4);
```
Comment (photodow) [inline packages/styles/scss/patterns/sub-patterns/content-group/_content-group.scss]: small tweak here. The CTA is 4 of 8. Same value here, but at the md breakpoint the grid is now 8.

File: packages/styles/scss/patterns/sub-patterns/content-group/_content-group.scss [hunk 20]
```diff
@@ -22,31 +20,40 @@
...
+      }
+      @include carbon--breakpoint('lg') {
+        @include carbon--make-col(4, 8);
+      }
```
Comment (photodow) [inline packages/styles/scss/patterns/sub-patterns/content-group/_content-group.scss]: This isn't doing anything for us, just adding extra css

File: packages/styles/scss/patterns/sub-patterns/content-group/_content-group.scss [hunk 21]
```diff
@@ -22,31 +20,40 @@
...
+
+      margin-bottom: $carbon--layout-03;
```
Comment (photodow) [inline packages/styles/scss/patterns/sub-patterns/content-group/_content-group.scss]: Why is this here? Is this supposed be attached to the CTA, or apart of the group container?

File: packages/styles/scss/patterns/sub-patterns/content-group/_content-group.scss [hunk 22]
```diff
@@ -22,31 +20,40 @@
...
       @include carbon--type-style('expressive-heading-04', true);
     }
 
     &__children {
-      padding-left: 0;
-      padding-right: 0;
+      margin-bottom: $carbon--layout-03;
+    }
+
+    &__CTA {
```
Comment (photodow) [inline packages/styles/scss/patterns/sub-patterns/content-group/_content-group.scss]: The card should be hanging into the gutter so the text aligns.

Comment (photodow) [inline packages/styles/scss/patterns/sub-patterns/content-group/_content-group.scss]: Also, the card is going well beyond the 320px max-width. (this maybe something you can apply to the card. I thought it was already applied for something)

File: packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js (hunk 16 above)
Comment (photodow) [reply packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: I think it's working, but I think it's working based on possibly legacy stuff within the card? Does it work if we change it to a text type link? Do you have the latest work from Anna's card PR?

File: packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js (hunk 12 above)
Comment (photodow) [reply packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: On that topic is there a reason we're reassigning it to another variable `customClassName`?

Comment (mkothur) [reply packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: if customeClassName is not passed so it will hadle, folowwed other components has the same way.

File: packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js (hunk 14 above)
Comment (mkothur) [reply packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: yeah sorry.

File: packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js (hunk 15 above)
Comment (mkothur) [reply packages/react/src/patterns/sub-patterns/ContentGroup/ContentGroup.js]: sure

File: packages/styles/scss/patterns/sub-patterns/content-group/_content-group.scss (hunk 22 above)
Comment (mkothur) [reply packages/styles/scss/patterns/sub-patterns/content-group/_content-group.scss]: Okay

Comment (mkothur) [reply packages/styles/scss/patterns/sub-patterns/content-group/_content-group.scss]: do I need to add max-width for CTA?

**Comment by mkothur** [comment]
@photodow: I updated the PR for above comments, couple comments I am not understanding.
